tests/components/aemet/test_config_flow.py

```diff
@@ -0,0 +1,87 @@
+"""Define tests for the AEMET OpenData config flow."""
+
+from unittest.mock import MagicMock, patch
+
+import requests_mock
+
+from homeassistant import data_entry_flow
+from homeassistant.components.aemet.const import DOMAIN
+from homeassistant.config_entries import ENTRY_STATE_LOADED, SOURCE_USER
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+
+from .util import aemet_requests_mock
+
+from tests.common import MockConfigEntry
+
+CONFIG = {
+    CONF_NAME: "aemet",
+    CONF_API_KEY: "foo",
+    CONF_LATITUDE: 40.30403754,
+    CONF_LONGITUDE: -3.72935236,
+}
+
+
+async def test_form(hass):
+    """Test that the form is served with valid input."""
+
+    with patch(
+        "homeassistant.components.aemet.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "homeassistant.components.aemet.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry, requests_mock.mock() as _m:
+        aemet_requests_mock(_m)
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}
+        )
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+        assert result["step_id"] == SOURCE_USER
+        assert result["errors"] == {}
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], CONFIG
+        )
+
+        await hass.async_block_till_done()
+
+        conf_entries = hass.config_entries.async_entries(DOMAIN)
+        entry = conf_entries[0]
+        assert entry.state == ENTRY_STATE_LOADED
+
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == CONFIG[CONF_NAME]
+        assert result["data"][CONF_LATITUDE] == CONFIG[CONF_LATITUDE]
+        assert result["data"][CONF_LONGITUDE] == CONFIG[CONF_LONGITUDE]
+        assert result["data"][CONF_API_KEY] == CONFIG[CONF_API_KEY]
+
+        assert len(mock_setup.mock_calls) == 1
+        assert len(mock_setup_entry.mock_calls) == 1
+
+        entry = MockConfigEntry(domain=DOMAIN, unique_id="aemet_unique_id", data=CONFIG)
```

Comment:
If we're going to test the abort case in this test too, we don't need the MockConfigEntry.

Reply:
This isn't using the correct unique_id either.

Reply:
Then I will revert my commit and go back to the separate tests, but I don't really know what was wrong after your changes @MartinHjelmare

Reply:
The unique id isn't correct I think. It should be the mocked coordinates.